crates/uv/src/commands/toolchain/list.rs

```diff
@@ -38,27 +53,63 @@ pub(crate) async fn list(
         .into_iter()
         .flatten();
 
-    let installed = {
-        InstalledToolchains::from_settings()?
-            .init()?
-            .find_all()?
-            .collect_vec()
-    };
+    let installed = find_toolchains(
+        &ToolchainRequest::Any,
+        SystemPython::Required,
+        &ToolchainSources::All(PreviewMode::Enabled),
+        cache,
+    )
+    // Raise any errors encountered during discovery
+    .collect::<Result<Vec<Result<Toolchain, ToolchainNotFound>>, DiscoveryError>>()?
+    .into_iter()
+    // Then drop any "missing" toolchains
+    .filter_map(|result| match result {
+        Ok(toolchain) => Some(toolchain),
+        Err(_) => None,
+    });
 
-    // Sort and de-duplicate the output.
     let mut output = BTreeSet::new();
     for toolchain in installed {
+        let kind = if matches!(toolchain.source(), ToolchainSource::Managed) {
+            Kind::Managed
+        } else {
+            Kind::System
+        };
         output.insert((
-            toolchain.python_version().version().clone(),
-            toolchain.key().to_owned(),
+            toolchain.python_version().clone(),
+            toolchain.os().to_string(),
+            toolchain.key().clone(),
+            kind,
+            Some(toolchain.interpreter().sys_executable().to_path_buf()),
```

Comment:
Oh okay, I see now. The significance of the path in the output is that it indicates a toolchain that is actually installed? But any entry _without_ a path means it is merely available and not installed?

The other thing that I'm a little confused by in the output:

```
❯ uv toolchain list --all-versions
cpython-3.12.3-macos-aarch64-none	/opt/homebrew/opt/python@3.12/bin/python3.12
cpython-3.12.3-macos-aarch64-none	/Users/zb/Library/Application Support/uv/toolchains/cpython-3.12.3-macos-aarch64-none/install/bin/python3
cpython-3.12.3-macos-aarch64-none
```

Here, `cpython-3.12.3-macos-aarch64-none` is repeated three times. I get the first two: the output makes it clear they are two distinct installations on my system. But what's the third? I guess it means it's available but not installed? But I have two of them installed already?

Reply:
> The significance of the path in the output is that it indicates a toolchain that is actually installed? But any entry without a path means it is merely available and not installed?

Yes

> But what's the third? I guess it means it's available but not installed? But I have two of them installed already?

Ah we shouldn't list the download there, that's a bug.